File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/interactive/zaap/TeleportRequestMessage.scala





















// Generated on 05/03/2013 10:34:52
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait TeleportRequestMessage extends MessageDeserializer {
  val messageId = 5961

  protected trait Message extends BaseMessage with Serializable {
    val messageId = TeleportRequestMessage.messageId

    val teleporterType: Byte
    val mapId: Int

    def serialize(buf: Buffer) {
      buf.writeByte(teleporterType)
      buf.writeInt(mapId)
    }

  }

  protected def create(teleporterType: Byte, mapId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val teleporterType = buf.readByte()
    val mapId = buf.readInt()

    create(teleporterType, mapId)
  }

}

object TeleportRequestMessage extends TeleportRequestMessage {
  case class MessageImpl(teleporterType: Byte, mapId: Int) extends Message
  type Target = MessageImpl

  override def create(teleporterType: Byte, mapId: Int) = MessageImpl(teleporterType, mapId)

  def apply(teleporterType: Byte, mapId: Int) = create(teleporterType, mapId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/TeleportRequestMessage.java


// Generated on 05/08/2013 19:37:53
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class TeleportRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5961;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte teleporterType;
    public int mapId;
    
    public TeleportRequestMessage() { }
    
    public TeleportRequestMessage(byte teleporterType, int mapId) {
        this.teleporterType = teleporterType;
        this.mapId = mapId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(teleporterType);
        buf.writeInt(mapId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        teleporterType = buf.readByte();
        if (teleporterType < 0)
            throw new RuntimeException("Forbidden value on teleporterType = " + teleporterType + ", it doesn't respect the following condition : teleporterType < 0");
        mapId = buf.readInt();
        if (mapId < 0)
            throw new RuntimeException("Forbidden value on mapId = " + mapId + ", it doesn't respect the following condition : mapId < 0");
    }
    
}
